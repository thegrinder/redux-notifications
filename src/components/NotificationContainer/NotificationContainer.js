import React, { useLayoutEffect, useRef, useState } from 'react';
import PropTypes from 'prop-types';
import Notification from '../Notification/Notification';

const propTypes = {
  children: PropTypes.node.isRequired,
  position: PropTypes.array.isRequired,
  isVisible: PropTypes.bool.isRequired,
  animationDuration: PropTypes.number.isRequired,
  animationEasing: PropTypes.string.isRequired,
  slideFromSide: PropTypes.string.isRequired,
};

const NotificationContainer = ({
  children,
  position,
  isVisible,
  animationDuration,
  animationEasing,
  slideFromSide,
}) => {
  const notificationRef = useRef(null);
  const [height, updateHeight] = useState(0);

  useLayoutEffect(() => {
    if (notificationRef.current) {
      updateHeight(notificationRef.current.clientHeight);
    }
  }, [notificationRef]);

  return (
    <Notification
      ref={notificationRef}
      position={position}
      notificationHeight={height}
      isVisible={isVisible}
      slideFromSide={slideFromSide}
      animationEasing={animationEasing}
      animationDuration={animationDuration}
    >
      {children}
    </Notification>
  );
};

NotificationContainer.propTypes = propTypes;

export default NotificationContainer;
