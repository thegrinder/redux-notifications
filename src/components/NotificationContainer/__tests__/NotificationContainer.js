import React from 'react';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

import NotificationContainer from '../NotificationContainer';

const children = <p>test</p>;
const props = {
  position: ['20px', '30px', 'auto', 'auto'],
  isVisible: true,
  animationDuration: 100,
  animationEasing: 'ease',
  slideFromSide: 'left',
};

const renderComponent = () =>
  render(<NotificationContainer {...props}>{children}</NotificationContainer>);

describe('<NotificationsContainer />', () => {
  afterEach(cleanup);

  it('should render correctly', () => {
    const {
      container: { firstChild },
    } = renderComponent();
    expect(firstChild).toBeDefined();
    expect(firstChild).toMatchSnapshot();
  });

  it('should have children', () => {
    const {
      container: { firstChild },
      getByText,
    } = renderComponent();
    const childrenElement = getByText('test');
    expect(firstChild).toContainElement(childrenElement);
  });
});
