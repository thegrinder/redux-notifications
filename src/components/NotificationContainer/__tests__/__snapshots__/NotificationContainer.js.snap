// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NotificationsContainer /> should render correctly 1`] = `
<div
  class="sc-AxjAm hvaknr"
>
  <p>
    test
  </p>
</div>
`;
